Add deliverytag to Message type

lawrencegripper opened the pull request:
This PR add the deliveryTag onto the Message type so it can be used by consumers of the library. 

I couldn't see an obvious unit test for this change so I added an additional check the integration tests to validate the field is populated and had a value the size I was expecting back from SB. 

See #122

vcabbage commented on client.go:
```diff
@@ -1086,6 +1086,10 @@ func (l *link) muxReceive(fr performTransfer) error {
+
+		if fr.DeliveryTag != nil {
+			l.msg.DeliveryTag = fr.DeliveryTag
+		}
```
The nil check can be omitted since there's no dereference.

lawrencegripper replied:
Good point!

vcabbage commented on integration_test.go:
```diff
@@ -163,6 +163,16 @@ func TestIntegrationRoundTrip(t *testing.T) {
+							receiveErr = fmt.Errorf("Error after %d receives: deliverytag should be 16 length byte array representing a UUID. Got: %v", i, err, len(msg.DeliveryTag))
+							return
+						}
+
```
The number of arguments doesn't match the formatters in the two calls to `fmt.Errorf` here. Looks like it's just the extraneous `err` arguments.

lawrencegripper replied:
Sorry could of silly mistakes here, will fix up.

vcabbage commented on types.go:
```diff
@@ -1644,6 +1644,9 @@ type Message struct {
 
+	// The DeliveryTag can be up to 32 octets of binary data
+	DeliveryTag []byte
+
```
Please add a period to the end of the sentence.

lawrencegripper commented:
Thanks for the review :+1:  Happy to add in the ability to set too and fix up the rest. Hopefully have changes back to you shortly.

lawrencegripper commented:
I've added in the send capability and fixed up the feedback. Let me know what you think, happy to fix up anything as needed. 

I saw the integration tests flapping a bit while running them and found that increasing some of the sleeps improved their reliability - hope that's alright.

vcabbage commented on client.go:
```diff
@@ -377,10 +381,15 @@ func (s *Sender) send(ctx context.Context, msg *Message) (chan deliveryState, er
+		s.nextDeliveryTag++
+	} else {
+		deliveryTag = msg.DeliveryTag
+	}
```
Minor style thing, default to `msg.DeliveryTag` then check if the slice is empty, avoiding the else block.

``` go
deliveryTag := msg.DeliveryTag
if len(deliveryTag) == 0 {
    ...
}
```

lawrencegripper replied:
Much cleaner, will fix.
